sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.students.controller.getAssessmentNOA.e2e-spec.ts

```diff
@@ -172,6 +173,150 @@ describe("AssessmentStudentsController(e2e)-getAssessmentNOA", () => {
       .expect(expectation);
   });
 
+  it("Should get the student NOA details for a full-time application when the application assessment is created.", async () => {
+    // Arrange
+    const enrolmentDate1 = addDays(1);
+    // Create the new student to be mocked as the authenticated one.
+    const student = await saveFakeStudent(db.dataSource);
+    // Mock user services to return the saved student.
+    await mockUserLoginInfo(appModule, student);
+    // MSFAA required for the NOA be returned.
+    const msfaaNumber = createFakeMSFAANumber(
+      { student },
+      {
+        msfaaState: MSFAAStates.Signed,
+      },
+    );
+    await db.msfaaNumber.save(msfaaNumber);
+    // Creates the applications to get the NOA.
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      {
+        msfaaNumber,
+        student,
+      },
+      {
+        offeringIntensity: OfferingIntensity.fullTime,
+      },
+    );
+    const assessmentData = {
+      exceptionalCosts: 330,
+      transportationCost: 1022.4,
+      partnerStudentLoans: 33,
+      secondResidenceCost: 7497,
+      alimonyOrChildSupport: 44,
+      parentalVoluntaryContribution: 40,
+      parentalDiscretionaryContribution: null,
+    } as Assessment;
+    const workflowData = {
+      calculatedData: {
+        returnTransportationCost: 100,
+        totalAdditionalTransportationAllowance: 120,
+      },
+    } as WorkflowData;
+    // Creates a new application's current assessment.
+    const newCurrentAssessment = await db.studentAssessment.save(
+      createFakeStudentAssessment(
+        {
+          auditUser: student.user,
+          application,
+          offering: application.currentAssessment.offering,
+          applicationEditStatusUpdatedBy: student.user,
+        },
+        {
+          initialValue: {
+            triggerType: AssessmentTriggerType.RelatedApplicationChanged,
+            assessmentData,
+            workflowData,
+          },
+        },
+      ),
+    );
+    application.currentAssessment = newCurrentAssessment;
+    await db.application.save(application);
```

Comment:
I couldn't follow the reason why we need a new assessment and disbursement here. Looks like it came over from the existing E2E test. I am not going to consider this as a blocker. But I would advise to avoid data creation that is not required or create data closely to test requirement for future E2E tests.